web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -73,6 +75,17 @@ export function Resources() {
     fetchInitial();
   }, [clusterId, search]);
 
+  const noResults =
+    attempt.status === 'success' && fetchedData.agents.length === 0;
+
+  const [isSearchEmpty, setIsSearchEmpty] = useState(true);
+
+  // Using a useEffect for this prevents the "Add your first resource" component from being
+  // shown for a split second when making a search after a search that yielded no results.
+  useEffect(() => {
+    setIsSearchEmpty(!params?.query && !params?.search);
+  }, [params.query, params.search]);
+
```

Comment:
Can't this be derived on render without a `useEffect`?

```js
const isSearchEmpty = !params?.query && !params?.search
```

Reply:
This was my initial approach, however what would happen is if you made a search that yielded no results, and then made another search, the "Add your first resource" screen would render for a split second before the results were shown

Reply:
Oh i see whats happening. I was understanding params as the search input, but it's actually the url params. and that doesn't happen until after the fetch.

Ideally, we'd have this effect happen as a result of the fetch, rather than after what's happened from that fetch (the url params updating) Can we move this into the `useInfiniteScroll` hook, update it in the `fetchInitial` and `fetchMore` methods, and then export it from the hook?

Reply:
After thinking about it more, this probably isn't the worst place to have it the way you have it now. Technically, this state is only telling us if the search is empty and not directly related to _actually searching_. So, up to you.

Reply:
i actually don't see what's happening... (i had same initial question as michael). why can we only determine the url filter params after fetching?

Reply:
If we update on every render, when you make another search, there is a split second before `attempt.status` updates from `success` to `processing`. During this split second, you see this flash on the screen before the loading indicator is visible:
![image](https://github.com/gravitational/teleport/assets/56373201/7bc94d33-47ab-4492-9993-67b8ec200e37)